torchgeo/datasets/advance.py

```diff
@@ -0,0 +1,223 @@
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.files = self._load_files(self.root)
+        self.classes = sorted(set(f["cls"] for f in self.files))
+        self.class_to_idx: Dict[str, int] = {c: i for i, c in enumerate(self.classes)}
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        image = self._load_image(files["image"])
+        audio = self._load_target(files["audio"])
+        cls_label = self.class_to_idx[files["cls"]]
+        label = torch.tensor(cls_label, dtype=torch.long)  # type: ignore[attr-defined]
+        sample = {"image": image, "audio": audio, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self, root: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, audio, label
+        """
+        images = sorted(glob.glob(os.path.join(root, "vision", "**", "*.jpg")))
+        wavs = sorted(glob.glob(os.path.join(root, "sound", "**", "*.wav")))
+        labels = [image.split(os.sep)[-2] for image in images]
+        files = [
+            dict(image=image, audio=wav, cls=label)
+            for image, wav, label in zip(images, wavs, labels)
+        ]
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(self, path: str) -> Tensor:
+        """Load the target audio for a single image.
+
+        Args:
+            path: path to the target
+
+        Returns:
+            the target audio
+        """
+        array = wavfile.read(path)[1]
+        tensor: Tensor = torch.from_numpy(array)  # type: ignore[attr-defined]
+        tensor = tensor.unsqueeze(0)
+        return tensor
+
+    def _check_integrity(self) -> bool:
```

Comment:
I took a stab at modifying the dataset checks similarly to #138, however I think this will need a separate PR for `VisionDatasets`. There are a couple things I want to work through w/o holding adding this dataset in for now.